main.py

```diff
@@ -2,23 +2,90 @@
 #
 # Simple manager prototype for xqemu
 #
-from PyQt5.QtWidgets import QApplication, QDialog, QFileDialog, QMainWindow, QMessageBox
+from PyQt5.QtWidgets import QApplication, QDialog, QFileDialog, QMainWindow, QMessageBox, QTableWidgetItem, QAbstractItemView, QHeaderView
 from PyQt5.uic import loadUiType
 from PyQt5 import QtCore, QtGui
 from qmp import QEMUMonitorProtocol
+from glob import glob1
 import sys
 import os, os.path
 import json
 import subprocess
 import time
 import platform
+import csv
+import sys
 
 SETTINGS_FILE = './settings.json'
+LIBRARY_CACHE = './library.cache'
 
 # Load UI files
 settings_class, _ = loadUiType('settings.ui')
 mainwindow_class, _ = loadUiType('mainwindow.ui')
 
+class DiscImageFormat(object):
+	XISO = 'XISO'
+	ISO9660 = 'ISO9660'
+	Unknown = 'Unknown format'
+
+def read_from_offset(file, offset, length):
+	file.seek(offset)
+	return file.read(length).decode('unicode-escape')
+
+def getDiscFileFormat(filepath):
+	file = open(filepath, 'rb')
+
+	#XISO/XGD
+	if read_from_offset(file, 0x10000, 20) == 'MICROSOFT*XBOX*MEDIA': return DiscImageFormat.XISO
+
+	# ISO 9660
+	for offset in [0x9001, 0x8801, 0x8001]:
+		if read_from_offset(file, offset, 5) == 'CD001': return DiscImageFormat.ISO9660
+
+	return DiscImageFormat.Unknown
+
+class CacheManager(object):
+	def __init__(self):
+		self.reset()
+
+	def reset(self):
+		self.cache = []
+
+	def save(self):
+		# Some python version stuff to get rid of empty lines in the cache
+		if sys.version_info[0] == 3:
+			cachefile = open(LIBRARY_CACHE, 'w', newline='')
+		else:
+			cachefile = open(LIBRARY_CACHE, 'wb')
+
+		with cachefile as csvfile:
+			cachewriter = csv.writer(csvfile)
+			for entry in self.cache:
+				cachewriter.writerow(entry)
+
+	def load(self):
+		if os.path.exists(LIBRARY_CACHE):
+			with open(LIBRARY_CACHE, 'r') as csvfile:
+				cachereader = csv.reader(csvfile, delimiter=',', quotechar='"')
+				for row in cachereader:
+					self.cache.append(row)
+		else:
+			self.reset()
+
+	def checkExisting(self, path):
+		baseName = os.path.basename(path)
+		filename, _ = os.path.splitext(baseName)
+		lastmodtime = os.path.getmtime(path)
+		size = os.path.getsize(path)
+		for entry in self.cache:
+			if len(entry) != 4: continue
+			if entry[0] == filename and entry[1] == str(round(lastmodtime, 0)) and entry[2] == str(size):
```

Comment:
This rounding stuff looks a bit bad in my opinion. Why do it everytime when storing / loading cache, instead of avoiding direct calls which cause this issue in the first place?

In other words, why not wrap:

```
baseName = os.path.basename(path)
filename, _ = os.path.splitext(baseName)
lastmodtime = os.path.getmtime(path)
size = os.path.getsize(path)
```

in a function, which generates a cache element? You already sort-of do this in `cacheFile()` it seems, but you don't re-use it properly or fix these rounding issues.